License headers:

include/uapi/linux/futex.h
/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */

kernel/futex.c:
// SPDX-License-Identifier: GPL-2.0-or-later

--- a/include/uapi/linux/futex.h.orig	2021-09-22 02:22:38.201653858 -0700
+++ b/include/uapi/linux/futex.h	2021-09-22 02:23:26.067655435 -0700
@@ -22,7 +22,7 @@
 #define FUTEX_WAIT_REQUEUE_PI	11
 #define FUTEX_CMP_REQUEUE_PI	12
 #define FUTEX_LOCK_PI2		13
-#define FUTEX_WAIT_MULTIPLE	14
+#define FUTEX_WAIT_MULTIPLE	31
 
 #define FUTEX_PRIVATE_FLAG	128
 #define FUTEX_CLOCK_REALTIME	256
--- a/kernel/futex.c.orig	2021-09-22 02:22:51.401654293 -0700
+++ b/kernel/futex.c	2021-09-22 02:25:09.279658836 -0700
@@ -4048,7 +4048,7 @@ futex_init_timeout(u32 cmd, u32 op, stru
 		return -EINVAL;
 
 	*t = timespec64_to_ktime(*ts);
-	if (cmd == FUTEX_WAIT)
+	if (cmd == FUTEX_WAIT || cmd == FUTEX_WAIT_MULTIPLE)
 		*t = ktime_add_safe(ktime_get(), *t);
 	else if (cmd != FUTEX_LOCK_PI && !(op & FUTEX_CLOCK_REALTIME))
 		*t = timens_ktime_to_host(CLOCK_MONOTONIC, *t);
